src/bin/mrtg

```diff
@@ -675,17 +675,19 @@ sub getcurrent {
 	# (OID:community@host or OID) that may have been specified with the RouterUptime
 	# target keyword
 	if( defined $rcfg->{ routeruptime }{ $rou } ) {
-		my( $noid, $nloc ) = split( /:/, $rcfg->{ routeruptime }{ $rou }, 2 );
-		# If only location (community@host) was specified then
-		# move the location details into the right place
-		if( $noid =~ /@/ ) {
-			$nloc = $noid;
-			$noid = undef;
-		}
-		# If no OID (community@host) was specified use the hardcoded default
-		if( not $noid ) {
-			$noid = 'sysUptime';
-		}
+        my ( $noid, $comm );
+        #break apart at the right most at symbol
+		my( $noidcomm, $nloc ) = $rcfg->{ routeruptime }{ $rou } =~ /^(.*)@([^@]+)$/;
+		debug("base","noidcom $noidcomm and nloc $nloc");
+        #now assess for presence of custom OID for uptime
+        if ($noidcomm =~ /:/ ) {
+            ( $noid, $comm ) = split( /:/, $noidcomm, 2);
+        }
+        else {
+            $noid='sysUptime';
+            $comm=$noidcomm;
+        }
+        debug("base","noid is $noid and comm is $comm");
```

Comment:
```suggestion
                my ( $noid, $comm );
                #break apart at the right most at symbol
		my( $noidcomm, $nloc ) = $rcfg->{ routeruptime }{ $rou } =~ /^(.*)@([^@]+)$/;
		debug("base","noidcom $noidcomm and nloc $nloc");
                #now assess for presence of custom OID for uptime
                if ($noidcomm =~ /:/ ) {
                    ( $noid, $comm ) = split( /:/, $noidcomm, 2);
                 }
                else {
                    $noid='sysUptime';
                    $comm=$noidcomm;
                }
                debug("base","noid is $noid and comm is $comm");
```

Reply:
whoops, sorry about the indent levels. That was going to be another thing I was hoping to provide fixes for - code formatting to make life nicer in code viewers/editors.